src/components/categories/add-category.tsx

```diff
@@ -1,17 +1,37 @@
+"use client";
 import React from "react";
 import { Dialog, DialogContent, DialogTrigger } from "../ui/dialog";
 import { Button } from "../ui/button";
 import CategoryForm from "./category-form";
+import { createCategory } from "../providers/database/category";
+import { toast } from "../ui/use-toast";
+import { Category } from "../../lib/types/config";
 
 const AddCategory = () => {
+  const handleOnAddCategory = async (values: Category, form: any) => {
```

Comment:
No need to use `try catch` and `async await` if using `.then` to handle promises.
Otherwise:
```ts
  const handleOnAddCategory = async (values: Category, form: any) => {
    try {
      await createCategory({
        ...values,
      });
      toast({
          description: "Added successfully",
      });
      form.reset();
    } catch (error: any) {
      toast({
        description: error.response.message,
      });
    }
  };
```